src/test/ui/repeat_count.stderr

```diff
@@ -42,12 +42,20 @@ error[E0308]: mismatched types
    |
 LL |     let f = [0; -4_isize];
    |                 ^^^^^^^^ expected usize, found isize
+help: you can convert an `isize` to `usize` or panic if it the converted value wouldn't fit
+   |
+LL |     let f = [0; (-4_isize).try_into().unwrap()];
+   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error[E0308]: mismatched types
   --> $DIR/repeat_count.rs:28:23
    |
 LL |     let f = [0_usize; -1_isize];
    |                       ^^^^^^^^ expected usize, found isize
+help: you can convert an `isize` to `usize` or panic if it the converted value wouldn't fit
+   |
+LL |     let f = [0_usize; (-1_isize).try_into().unwrap()];
```

Comment:
for now ok, but we should try to not suggest code that is known to panic

Reply:
This is indeed a bad corner case :-/

Reply:
Filed https://github.com/rust-lang/rust/issues/60384